### ISSUE TITLE
Refine modals

### ISSUE THREAD

**Issue by staab** [coracle-social/coracle#482]
The bottom drawer is over-used, and sort of ugly when the screen is wide. First, refine it so that it is offset from the left, leaving the sidebar accessible. Next, add a regular modal dialog version of the component and use it everywhere except:

- HUD
- Profile
- Feed Edit

Don't create a new component, just add a parameter to modals that opts-in to drawer, and do an if/else in the modal component. There's too much delicate history-related stuff there we don't want to mess up.

**Comment by ticruz38** [comment]
For mobile it would make sense to keep the bottom up slide rather than from the left?

Also make it dismissible with a swap, something like the vaul-svelte library

**Comment by staab** [comment]
Yeah, mobile could probably keep the drawer as-is. vaul-svelte is nice, but I'm reluctant to add a dependency like that. Let me think about it.

**Comment by ticruz38** [comment]
No need to use vaul-svelte, if just for the swap to dismiss. Vaul-svelte is applying some more 3D transformation that can be heavy on mobile

**Comment by staab** [comment]
Hmm it seems I wasn't specific enough. The current bottom drawer is what should be offset to keep the nav visible. The new modal shouldn't be a drawer coming in from the left, but a more traditional modal (like exists in flotilla).

<img width="740" alt="Screenshot 2024-12-02 at 9 07 50 AM" src="https://github.com/user-attachments/assets/c5d6e85e-f4b6-4df7-8f22-fccfa724a917">

**Comment by ticruz38** [comment]
The new modal:

For posts with no replies yet, the modal will adjust to its own height.

https://github.com/user-attachments/assets/67f82df4-b9ba-49f4-a0bc-0dd974889118

For posts with more content, it will almost take up the full screen height

https://github.com/user-attachments/assets/3ea5fbd7-78b9-4d20-9e32-ca267f11ccec

The drawer refined:

Takes just the width to display a feed, leave the desktop menu visible

https://github.com/user-attachments/assets/210a2fd2-8960-451f-94c3-7bc955c11d56

**Comment by staab** [comment]
Cool, some small tweaks:

- Make the margins on the top/bottom/sides of the modal equal
- Make the drawer span the full content width (the light border you had before seemed good)

@dtonon any other input?

**Comment by ticruz38** [comment]
Padding on drawer and modal are the same fine border now

<img width="1728" alt="Screenshot 2024-12-04 at 17 16 44" src="https://github.com/user-attachments/assets/99ebe065-2c39-4240-8c6b-9c56e019041f">
<img width="1728" alt="Screenshot 2024-12-04 at 17 16 33" src="https://github.com/user-attachments/assets/c24a2eef-42ec-404a-b10d-f20bc34b63fc">

**Comment by dtonon** [comment]
> @dtonon any other input?

Centering the bottom sheet is definitely a big step forward, good work @ticruz38!

But I would keep the background's width of the lighbox wide as much as the available space (minus the left sidebar).
It's better to keep the background dark(er) like the main view, so we can avoid this backgrounds' chromatic flipping, that it's quite disturbing (at least for me) and overcomplicate the design without any real benefit.

Form reference, my proposed design:
(don't mind the double sidebar, the concept is the same)

<img width="1048" alt="image" src="https://github.com/user-attachments/assets/288c246c-1326-467e-812a-1bc9ded9f94c">

**Comment by ticruz38** [comment]
@dtonon 
I have 2 questions:
- In your design, there is no real lightbox around the modal content, the modal content is immediately superposed to the modal backdrop? 
- Should the backdrop fade in to the colour of the sidebar? What is the overall open animation for this modal?

As a side note - the dark mode atm has a lighter sidebar background than the app content background, this modal design requires me to flip the 2 colours

**Comment by dtonon** [comment]
@ticruz38 the lightbox fills only the content of the right panel, leaving the left sidebar(s) accessible. I'm not sure to understand what you mean by 

> the modal content is immediately superposed to the modal backdrop?

but I think yes, is what I envisioned.

For the foreground we have two options: darken the modal background a bit , like in the mockup; or keep the same color of the main view's background, and use a darker semi-transparent backdrop, so the top will result a little darker.
I prefer my proposal since this way the top content is more contrasted.
Keep in mind that in my general idea we should only have one bottom-sheet modal, so there is not the problem of stacked modals.

The animation should be like the current one, a bottom-up movement.

> As a side note - the dark mode atm has a lighter sidebar background than the app content background, this modal design requires me to flip the 2 colours

Why? You can just use the same colors of the main view, darkening a little the background as explained before.

Let me know if you need a review of the design's visual to have a better feel of the overall proposal.

**Comment by ticruz38** [comment]
@dtonon except for the close button that is still outside the modal, is the following recording aligned with you idea (colour & animation)?


https://github.com/user-attachments/assets/f7b9f52a-a6af-4c1c-b685-2f378f7045b1

Once the buttons are inside the modal, the idea would be to leave enough space for the top nav to be visible under the backdrop.

**Comment by dtonon** [comment]
@ticruz38 sorry for the delay, the overall effect is fine; the only difference are the colors: my proposal keep the main background dark with a slight lighter background for the notes. I would keep this combination, without flipping the colors, even with the current multi-modal structure. @staab is color flipping necessary for anything else?

**Comment by staab** [comment]
No, the color flipping is a very old thing, probably pre-dating the dark backdrop. I think we could use the same color palette as page content and it would be fine. (color alternation is still needed for nested quotes)

**Comment by dtonon** [comment]
> color alternation is still needed for nested quotes

I would switch to a simple border keeping the main background color.
It's visually cleaner, more readable and could also allow nested mentions without much troubles.

**Comment by staab** [comment]
Like this? I dunno... ![Screenshot 2024-12-18 at 8 53 59 AM](https://github.com/user-attachments/assets/325159cd-5ef6-4cb8-87e1-afb078a95e8b)

**Comment by dtonon** [comment]
@staab sorry for the delay (damn missed notifications), yes something like this, but with a bolder border and darker color.
Example:

<img width="823" alt="Screenshot 2025-01-14 at 17 23 55" src="https://github.com/user-attachments/assets/73a01a1c-f29f-49e2-82ab-a30351283475" />
<br/><br/>
We can also play a little with colors:

<img width="850" alt="Screenshot 2025-01-14 at 17 19 12" src="https://github.com/user-attachments/assets/e7395bdc-78c2-4c23-b7cf-c7cb1bf42276" />
<br/><br/>
(this version misses the border-radius because border-image requires a different html structure to simulate the radius)

I would also add some space before the mention (in the example, 20px).

/cc @ticruz38
